### PR TITLE
Bug fixes for the failed tasks and pause API

### PR DESCRIPTION
**Pull request by saikaranam-amazon**
### Description
Bug fixes for the failed tasks and pause API
 
### Issues Resolved
N/A
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt [hunk 1]
```diff
@@ -627,13 +624,15 @@ open class IndexReplicationTask(id: Long, type: String, action: String, descript
         return MonitoringState
     }
 
-    private fun findAllReplicationFailedShardTasks(clusterState: ClusterState)
+    private fun findAllReplicationFailedShardTasks(followerIndexName: String, clusterState: ClusterState)
```
Comment (gbbafna) [inline src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt:627]: nit : we already have `followerIndexName` as a member .

File: src/main/kotlin/org/opensearch/replication/action/pause/TransportPauseIndexReplicationAction.kt [hunk 2]
```diff
@@ -71,9 +71,9 @@ class TransportPauseIndexReplicationAction @Inject constructor(transportService:
                 validatePauseReplicationRequest(request)
 
                 // Restoring Index can't be paused
-                val restoring = clusterService.state().custom<RestoreInProgress>(RestoreInProgress.TYPE).any { entry ->
+                val restoring = clusterService.state().custom<RestoreInProgress?>(RestoreInProgress.TYPE)?.any { entry ->
                     entry.indices().any { it == request.indexName }
-                }
+                } ?: false
```
Comment (gbbafna) [inline src/main/kotlin/org/opensearch/replication/action/pause/TransportPauseIndexReplicationAction.kt:76]: we have similar code in `transportStopReplication` as well which will need this fix .
btw, why do we need this fix ?

File: src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt (hunk 1 above)
Comment (saikaranam-amazon) [reply src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt:627]: Keeping as is for now

File: src/main/kotlin/org/opensearch/replication/action/pause/TransportPauseIndexReplicationAction.kt (hunk 2 above)
Comment (saikaranam-amazon) [reply src/main/kotlin/org/opensearch/replication/action/pause/TransportPauseIndexReplicationAction.kt]: We are hitting the issues if entry is not present. Updated the code

**Review by krishna-ggk — APPROVED** [review]

**Pull request merged**